llama_index/indices/managed/google/generativeai/base.py

```diff
@@ -187,7 +206,11 @@ def as_query_engine(
         """
         return super().as_query_engine(
```

Comment:
I think this won't actually work with `super()` ? 

Super method
```
def as_query_engine(self, **kwargs: Any) -> BaseQueryEngine:
    # NOTE: lazy import
    from llama_index.query_engine.retriever_query_engine import RetrieverQueryEngine

    retriever = self.as_retriever(**kwargs)

    kwargs["retriever"] = retriever
    if "service_context" not in kwargs:
        kwargs["service_context"] = self._service_context
    return RetrieverQueryEngine.from_args(**kwargs)
```

You can see the retriever will be overwritten. Maybe it's best to re-implement the method and avoid `super()`

Reply:
I think I lucked out here that it would work because super() uses self.as_retriever(**kwargs), which is exactly what I am passing too in line 189 below.

But I agree the code would probably read better if I just re-implement the method.

So, I'm done as you suggested!